### PR TITLE
5.3 formula with ice3.6 as default

### PR DESCRIPTION
**Pull request by jburel**
In this PR: 

Introduce  OMERO 5.3 formula
This formula will be installed with ice 3.6

```
brew install Formula/omero53.rb
```

### CODE REVIEW

**Comment by sbesson** [comment]
@jburel are you planning to integrate the changes from https://github.com/ome/homebrew-alt/pull/117 here and should I close this other PR?

**Comment by jburel** [comment]
@sbesson I will integrate yours changes in this PR. Thanks

**Comment by sbesson** [comment]
One general thought completely outside the scope of this PR as we are starting a new versioned formula. The core Homebrew scheme for versioned formulas has recently evolved to use a naming scheme with the `@` symbol e.g. `qt@5.7` (http://docs.brew.sh/Versions.html). We might investigate switching to this representation at some point in the future if we keep maintaining these.

**Comment by jburel** [comment]
With the various evolutions on the sysadmin front, I am not convinced that we want to keep maintain the formulae

**Comment by sbesson** [comment]
See https://ci.openmicroscopy.org/job/OMERO-5.2-merge-homebrew/ICE=3.6,OSX=snapper/399/console. With Ice 3.6 we also need an extra `pip install zeroc-ice` step in order to use the CLI/OMERO.web. If not possible to add via the formula directly, we might need to add this step to `omero-install` and/or the documentation accordingly

**Comment by jburel** [comment]
Ooops forgot to add that step. Will check

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-homebrew/ICE=3.6,OSX=snapper/411/ is green

**Comment by sbesson** [comment]
Overall the changes proposed here and in the corresponding https://github.com/ome/omero-install/pull/143 make sense and put us in a good position for OMERO 5.3.0. As discussed by voice, I would argue in favor of only merging this PR with a new formula once we have released a milestone including the final DB changes.

One question regarding the testing strategy: does the `omero52` formula need to be tested locally? Also do we want to test both Ice 3.5 and 3.6 to be tested via the homebrew job or limit the job to the recommended Ice version?

**Comment by jburel** [comment]
I will limit to the recommended version
We could do a tmp job to test the 5.2 change

File: Formula/omero53.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
+require 'formula'
+
+class Omero53 < Formula
+  homepage 'http://www.openmicroscopy.org/site/products/omero'
+
+  url 'http://downloads.openmicroscopy.org/omero/5.3.0-m8/artifacts/openmicroscopy-5.3.0-m8.zip'
+  sha256 '9cffaadceb9245748686774c971da7020f10dcb798ec63b28cb494f022f87fe6'
+
+  option 'with-cpp', 'Build OmeroCpp libraries.'
+  option 'with-ice36', 'Use Ice 3.6.'
```
Comment (sbesson) [inline Formula/omero53.rb]: Is that supposed to be `with-ice35` as per the dependency declaration below? At the moment, this current version of the formula might unconditionally build with Ice 3.6.

Comment (jburel) [inline Formula/omero53.rb]: should be 3.5

**Comment by will-moore** [comment]
I was trying to follow https://www.openmicroscopy.org/site/support/omero5.3-staging/sysadmins/unix/server-install-homebrew.html and upgrade from Ice 3.5 to Ice 3.6 but have managed to screw something up and now I don't have either working!

I don't see any equivalent of 
```
export PYTHONPATH=$(brew --prefix omero53)/lib/python
export ICE_CONFIG=$(brew --prefix omero53)/etc/ice.config
```
or 
```
pip install -r $(brew --prefix omero53)/share/web/requirements-py27-nginx.txt
bash bin/omero_python_deps
```
under the "Development Server" instructions. 

```
$ brew install --only-dependencies omero
==> Installing omero52 from ome/alt
All dependencies for ome/alt/omero52 are satisfied.
ls31618:OMERO wmoore$ which icegridnode
/usr/local/bin/icegridnode
ls31618:OMERO wmoore$ icegridnode -v
3.5.1
ls31618:OMERO wmoore$ python
Python 2.7.10 (default, Sep 23 2015, 04:34:14) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.72)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import Ice
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named Ice
```
Now I do
```
$ pip install "zeroc-ice>3.5,<3.7"
...
>>> import Ice
>>> Ice.stringVersion()
'3.6.3'
```
But I still have 
```
$ icegridnode -v
3.5.1
```
Do I just need to wait for this PR to be merged?

**Comment by will-moore** [comment]
Ah... after this...
```
$ brew install ice
$ brew link --overwrite ice
```
looking a bit better now...

**Comment by jburel** [comment]
@will-moore the formula is not available yet

**Comment by jburel** [comment]
remove comments, 52 formula changes are no longer part of this PR

**Pull request merged**